Utilities/wearable-packager/marketplace-packager.js

```diff
@@ -0,0 +1,144 @@
+//  for releasing to Marketplace as a wearable item.
+// 
+//  Created by Liv Erickson on 11/16/17
+//  Copyright 2017 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+/* globals Selection */
+
+(function(){
+    var APP_NAME = "WEARPKGR";
+    var APP_URL = Script.resolvePath("app.html");
+    var APP_ICON = Script.resolvePath("icon.png");
+
+    var prevID = 0;
+    var listName = "contextOverlayHighlightList";
+    var listType = "entity";
+
+    var entityIDToExport = "";
+
+    var tablet = Tablet.getTablet('com.highfidelity.interface.tablet.system');    
+
+    function handleMousePress(entityID) {
+        if (prevID !== entityID) {
+            Selection.addToSelectedItemsList(listName, listType, entityID);
+            prevID = entityID;
+        }
+        tablet.emitScriptEvent(entityID);
+    }
+
+    function handleMouseLeave(entityID) {
+        if (prevID !== 0) {
+            Selection.removeFromSelectedItemsList("contextOverlayHighlightList", listType, prevID);
+            prevID = 0;
+        }
+    }
+
+    var baseUserdata = {
+        "Attachment": {
+            "action": "attach",
+            "joint": "Hips",
+            "attached" : false,
+            "options": {
+                "translation": {
+                    "x": 0,
+                    "y": 0,
+                    "z": 0
+                },
+                "scale": 1
+            }
+        },
+        "grabbableKey": {
+            "cloneable": false,
+            "grabbable": true
+        }
+    };
+
+    var exportProperties = {
+        type: "Model",
+        clientOnly: 1,
+        parentID: "{00000000-0000-0000-0000-000000000001}",
+        owningAvatarID: "{00000000-0000-0000-0000-000000000000}",
+        visible: 1,
+        shapeType: "box",
+        collidesWith: "",
+        collisionMask: 0
+    };
+
+    var button = tablet.addButton({
+        text: APP_NAME,
+        icon: APP_ICON
+    });
+
+    
+    function maybeExited() {
+        Entities.clickReleaseOnEntity.disconnect(handleMousePress);
+        Entities.hoverLeaveEntity.disconnect(handleMouseLeave);
+        tablet.screenChanged.disconnect(maybeExited);
+    }
+
+    function clicked(){
+        tablet.gotoWebScreen(APP_URL);
+        Entities.clickReleaseOnEntity.connect(handleMousePress);
+        Entities.hoverLeaveEntity.connect(handleMouseLeave);
+        Script.setTimeout(function(){
+            tablet.screenChanged.connect(maybeExited); 
+        }, 2000);
+    }
+    button.clicked.connect(clicked);
+
+    function onFileSaveChanged(filename){
+        Window.saveFileChanged.disconnect(onFileSaveChanged);
+        if (filename !== "") {
+            var success = Clipboard.exportEntities(filename, [entityIDToExport]);
+            if (!success) {
+                // No luck, failed
```

Comment:
Perhaps show a Window.alert()  dialog or something similar to inform the user?

Reply:
Window.alert() in tablet apps is kind of messy. If we want to build this out into a fully designed app instead of a utility then I would agree